include/mbedtls/ssl_ciphersuites.h

```diff
@@ -292,6 +292,13 @@ typedef enum {
 #define MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED
 #endif
 
+/* Key exchanges in either TLS 1.2 or 1.3 which are using an ECDSA
+ * signature */
+#if defined(MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED) || \
+    defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED)
+#define MBEDTLS_KEY_EXCHANGE_WITH_ECDSA_ANY_ENABLED
+#endif
+
 #if defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED) || \
     defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED)
```

Comment:
Seems we have two separate macro guards for the cert-based `key exchange `and `client authentication `in TLS1.2, but for TLS1.3 we only one guard to protect them.   Is "guard them separately" better?

Reply:
Sorry, can you clarify which are the two macros fro 1.2 and the single macro for 1.3 that you are referring to?

The protocols are different regarding how authentication works, and I think having a single macro for 1.3 is correct, but now that you mention it, perhaps a single macro is enough for 1.2 as well. I you could kindly indicate the names of the macros you're referring to for each protocol, I'll double-check whether we need exactly those, or perhaps one more or one less.

Reply:
> can you clarify which are the two macros fro 1.2 and the single macro for 1.3 that you are referring to?

For TLS1.2, I just see MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED for the `cert based key exchange`, and MBEDTLS_KEY_EXCHANGE_ECDSA_CERT_REQ_ALLOWED_ENABLED) for the `cert based client authentication`.  And for TLS1.3, we always use MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED to guard both the cert based `key exchange `and `client authentication`.

Reply:
Thanks for the precisions! Looking at the macros are defined, I think this is correct and reflects differences the protocols:
- in TLS 1.3, all key exchange modes that allow a server certificate also allow a client certificate ("all" is actually just one mode: Ephemeral);

Reply:
I'm adding a comment to clarify.